ValkeyJSON.md

```diff
@@ -0,0 +1,214 @@
+which is getValues, setValues or deleteValues.
+
+Insert and update are implemented as a 2-phase operation that splits the write operation into two calls - prepareSetValues 
+and commit, where prepareSetValues does not change the Valkey data. The purpose of having a 2-phase write is to be able 
+to discard the write operation if certain conditions are not satisfied.
+
+### Tokenization of JSON Object Keys
+
+JSON has two container types - Array and Object. The JSON Object is a key/value mapping where keys are restricted to 
+be strings, whereas the value can be any valid JSON value, including another container value. Thus, a single JSON document 
+can contain multiple JSON objects, each with its own unique namespace. It’s quite common that a key repeatedly appears 
+in the same JSON document or across documents. If the number of JSON documents is large, repeated key names can consume 
+significant amount of memory usage.
+
+To remove duplicate copies of JSON object key names, the module tokenizes key names by maintaining a global data structure 
+called KeyTable, which is a sharded hash table storing key tokens and reference counts. Access to the KeyTable is threadsafe.
+
+## Specification
+
+### Supported JSON Standard
+
+[RFC7159](http://www.ietf.org/rfc/rfc7159.txt) and [ECMA-404](http://www.ecma-international.org/publications/standards/Ecma-404.htm) 
+JSON data interchange standard is supported. UTF-8 Unicode in JSON text is supported.
+
+### Root JSON Value
+
+In earlier RFC 4627, only objects or arrays were allowed as root values of JSON. Since [RFC 7159](http://www.ietf.org/rfc/rfc7159.txt), 
+the root value of a JSON document can be of any type, scalar type (String, Number, Boolean, Null) or container type (Array, Object). 
+ValkeyJSON is compliant with [RFC 7159](http://www.ietf.org/rfc/rfc7159.txt).
+
+### JSON Command API
+
+There are 22 JSON commands. See https://docs.aws.amazon.com/memorydb/latest/devguide/json-list-commands.html. The API is 
+compatible with RedisJSON v2. Note that API compatibility here means our command API is a superset of RedisJSON API. 
+For example, we have command “JSON.DEBUG DEPTH” and “JSON.DEBUG FIELDS”, while they do not.
+
+### JSONPath Query Syntax
+
+Two kinds of JSONPath query syntaxes are supported:
+* Enhanced syntax – Follows the [Goessner-style](https://goessner.net/articles/JsonPath/) JSONPath query syntax, as shown in the table below.
+* Restricted syntax – Has limited query capabilities, compatible with RedisJSON v1.
+
+If a query path starts with '$', the enhanced syntax is used. Otherwise, the restricted syntax is used. A query using 
+the enhanced syntax always returns an array of values, while a restricted-syntax query always returns a single value. 
+
+![](./images/jsonpath_syntax.png)
+
+For details and examples, see https://docs.aws.amazon.com/memorydb/latest/devguide/json-document-overview.html#json-path-syntax.
```

Comment:
See earlier comments regarding PNGs and references to AWS documentation.

Reply:
Removed the PNGs and references AWS documentation. It's now all self-contained markdown text.